tcmur_cmd_handler.c

```diff
@@ -2308,6 +2318,312 @@ static int handle_inquiry(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
+	};
+
+	static cdb_info opcode_extended_copy = {
+		.opcode = EXTENDED_COPY,
+		.cdb_length = 16,
+		.cdb_usage = {EXTENDED_COPY, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+		              0x00, 0x00, 0xff, 0xff, 0xff, 0xff, 0x00, CONTROL_BYTE},
+	};
+	
+	static cdb_info opcode_read_16 = {
+		.opcode = READ_16,
+		.cdb_length = 16,
+		.cdb_usage = {READ_16, 0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+		              0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, CONTROL_BYTE},
+	};
+
+	static cdb_info opcode_write_16 = {
+		.opcode = WRITE_16,
+		.cdb_length = 16,
+		.cdb_usage = {WRITE_16, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+		              0x00, 0x00, 0xff, 0xff, 0xff, 0xff, CONTROL_BYTE},
+	};
+
+	static cdb_info opcode_write_same_16 = {
+		.opcode = WRITE_SAME_16,
+		.cdb_length = 16,
+		.cdb_usage = {WRITE_SAME_16, 0xf8, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+		              0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, CONTROL_BYTE},
+	};
+
+	static cdb_info opcode_write_verify_16 = {
+		.opcode = WRITE_VERIFY_16,
+		.cdb_length = 16,
+		.cdb_usage = {WRITE_VERIFY, 0xf2, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+		              0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, CONTROL_BYTE},
+	};
+
+	static cdb_info opcode_format_unit = {
+		.opcode = FORMAT_UNIT,
+		.cdb_length = 6,
+		.cdb_usage = {FORMAT_UNIT, 0x00, 0xff, 0xff, 0xff, CONTROL_BYTE},
+	};
+
+	switch (op) {
+	case INQUIRY:
+		buf = opcode_inquiry.cdb_usage;
+		break;
+	case TEST_UNIT_READY:
+		buf = opcode_test_unit_ready.cdb_usage;
+		break;
+	case SERVICE_ACTION_IN_16:
+		buf = opcode_service_action_in_16.cdb_usage;
+		break;
+	case READ_CAPACITY:
+		buf = opcode_read_capacity.cdb_usage;
+		break;
+	case MODE_SENSE:
+		buf = opcode_mode_sense.cdb_usage;
+		break;
+	case MODE_SENSE_10:
+		buf = opcode_mode_sense_10.cdb_usage;
+		break;
+	case START_STOP:
+		buf = opcode_start_stop.cdb_usage;
+		break;
+	case MODE_SELECT:
+		buf = opcode_mode_select.cdb_usage;
+		break;
+	case MODE_SELECT_10:
+		buf = opcode_mode_select_10.cdb_usage;
+		break;
+	case RECEIVE_COPY_RESULTS:
+		buf = opcode_receive_copy_results.cdb_usage;
+		break;
+	case MAINTENANCE_IN:
+		switch (sa) {
+		case MI_REPORT_TARGET_PGS:
+			buf = opcode_mi_report_target_pgs.cdb_usage;
+			break;
+		case MI_REPORT_SUPPORTED_OPERATION_CODES: 
+			buf = opcode_mi_supported_operation_codes.cdb_usage;
+			break;
+		default:
+			break;
+		}
+		break;
+	case UNMAP:
+		buf = opcode_unmap.cdb_usage;
+		break;
+	case EXTENDED_COPY:
+		buf = opcode_extended_copy.cdb_usage;
+		break;
+	case READ_16:
+		buf = opcode_read_16.cdb_usage;
+		break;
+	case WRITE_16:
+		buf = opcode_write_16.cdb_usage;
+		break;
+	case WRITE_SAME_16:
+		buf = opcode_write_same_16.cdb_usage;
+		break;
+	case WRITE_VERIFY_16:
+		buf = opcode_write_verify_16.cdb_usage;
+		break;
+	case FORMAT_UNIT:
+		buf = opcode_format_unit.cdb_usage;
+		break;
+	default:
+		break;
+	}
+
+	return buf;
+}
+
+int is_support_opcode(int opcode) 
```

Comment:
Drop the extra space after the ending ")".

Also, I think it should be renamed to:

is_supported_opcode()